proto/asn1-pdu/x509_certificate.proto

```diff
@@ -134,13 +134,36 @@ message UniqueIdentifier {
   required asn1_universal_types.BitString value = 2;
 }
 
+message Extension {
+  // RFC 5280 4.2.1.1 & 4.2.1.2: Extensions such as |AuthorityKeyIdentifier| and
+  // |SubjectKeyIdentifier| must be marked as non-critical by the CA. However,
+  // allow it to be true to generate interesting inputs.
+  required asn1_universal_types.Boolean critical = 1;
+  oneof types {
+    RawExtension raw_extension = 2;
+    AuthorityKeyIdentifier authority_key_identifier = 3;
+    SubjectKeyIdentifier subject_key_identifier = 4;
+  }
+}
+
 // See RFC 5280, 4.1 & 4.1.2.9.
-message Extensions {
-  // If |pdu| is present, encode |Extensions| as an arbitraty pdu.
-  required asn1_pdu.PDU pdu = 1;
-  required asn1_pdu.PDU value = 2;
-  // TODO(dhalawi@gmail.com): switch the required
-  // field to encompass a valid |Extensions| value.
+message RawExtension {
+  required asn1_universal_types.ObjectIdentifier object_identifier = 1;
```

Comment:
```suggestion
  required asn1_universal_types.ObjectIdentifier extnID = 1;
```

(alternatively, `extension_id`)

Let's try to keep the protobuf field names similar to the ASN.1 field names, where appropriate, so that it's easier to map against X.509/RFC 5280.